### DISCUSSION TITLE
CD don't set working dir for buffers

### DISCUSSION THREAD

**Discussion by astephanh** [nvim-tree/nvim-tree.lua#2323] [Q&A]
### Description

Hi,

if i invoke change_root_to_node(), buffers still use the old working directory while the nvim-tree buffer itself has the new working directory. 

My Quick Fix for this: 

  -- change_root_to_node will not change pwd for other buffers,
  -- so i set it in this function
  local api = require("nvim-tree.api")

  vim.keymap.set("n", "<C-]>", function()
    local node = api.tree.get_node_under_cursor()
    api.tree.change_root_to_node()
    vim.api.nvim_set_current_dir(node.absolute_path)
  end, opts("CD"))

### Neovim version

```text
NVIM v0.9.1
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

Linux 6.3.12-1-MANJARO

### nvim-tree version

a708bd2

### Clean room replication

```lua
just invoke the funktion, then use

:pwd 

in Nvim-tree Buffer vs in other buffers
```

### Steps to reproduce

 just invoke the funktion, then use

:pwd 

in Nvim-tree Buffer vs in other buffers

**Comment by alex-courtis** [comment]
These options might enable the behaviour you desire:
`:help nvim-tree.sync_root_with_cwd`
`:help nvim-tree.respect_buf_cwd`

**Comment by astephanh** [comment]
Hi, 

if i'm not mistaken, this is not what i want. I use api.tree.change_root_to_node() to quickly change the Path between different directories in my Session for search / grep Operations (with Telescope find_files and other functions). But i want to change pwd without changing the root node for the tree. Does this makes sense? I was working that way in Nerdtree (and neo-tree as well if i remember correctly)

**Reply by alex-courtis** [reply]
> I use api.tree.change_root_to_node() to quickly change the Path between different directories in my Session for search / grep Operations (with Telescope find_files and other functions).

My approach is a little different: use an nvim-tree mapping to execute telescope etc. at the directory: https://github.com/alex-courtis/arch/blob/ee39f33434684f35520860e232bdce9717aa3702/config/nvim/lua/amc/plugins/nvt.lua#L34

**Reply by alex-courtis** [reply]
>  I use api.tree.change_root_to_node() to quickly change the Path

> But i want to change pwd without changing the root node for the tree.

I don't quite understand: change root to node will change the root.

Perhaps you could implement an nvim-tree mapping that changes global or local cwd temporarily e.g.

```lua
local function quickly_change()
  print(vim.cmd.pwd())

  local node = api.tree.get_node_under_cursor()
  if not node or node.type ~= "directory" then
    return
  end

  vim.cmd.cd(node.absolute_path)
  print(vim.cmd.pwd())

  -- do something in that directory

  vim.cmd.cd("-")
  print(vim.cmd.pwd())
end
```

```lua
  vim.keymap.set('n', 'c', quickly_change, opts('Quick Change'))
```

**Comment by tuyendinh** [comment]
> vim.keymap.set("n", "<C-]>", function() local node = api.tree.get_node_under_cursor() api.tree.change_root_to_node()  vim.api.nvim_set_current_dir(node.absolute_path) end, opts("CD"))

I have same issue as you, but for your case, I think you can try with setting 
     ``` actions = { 
        change_dir = {
          enable = true,
          global = true,
          restrict_above_cwd = false,
        },  ```
        this make directory change by :cd instead of :lcd 
       For me, I have multiple tabs, so I don't want to use :cd, I expect there should be a support with :tcd

**Reply by alex-courtis** [reply]
tcd may come with #2255
